tests/cmdline/commands/test_export.py

```diff
@@ -213,6 +201,27 @@ def test_migrate_force(self):
                 self.assertTrue(os.path.isfile(filename_output))
                 self.assertEqual(zipfile.ZipFile(filename_output).testzip(), None)
 
+    def test_migrate_in_place(self):
+        """Test that passing the -i/--in-place option will overwrite the passed file."""
+        archive = 'export_v0.1_simple.aiida'
+        target_version = '0.2'
+        filename_input = get_archive_file(archive, filepath=self.fixture_archive)
+        filename_tmp = next(tempfile._get_candidate_names())
+
+        try:
```

Comment:
why not use:
```
with tempfile.NamedTemporaryFile() as handle:
    filename = handle.name
```
That way no need to manually remove nor use protected `_get_candidate_names`

Reply:
I saw in the [docs](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) that 
> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms

In any case, I was a bit worried about opening a handle to a file that is then copied over, read, deleted and copied over again (perhaps it would be fine as well...)

Reply:
Since we're only supporting Unix-y platforms, that shouldn't be an issue though, right?

Reply:
> Since we're only supporting Unix-y platforms, that shouldn't be an issue though, right?

So far ;-) 
Maybe it never happens, but I believe circus is our only dependency currently that is not available on windows (at least it was some time ago), and fixes for windows are underway.

Perhaps it never happens... but let's keep the option open I would say.

Reply:
Not sure about the deleting and re-creating part, either.

Reply:
Anyway, if you want to avoid manually getting the name, you could create a `TemporaryDirectory` and then just name the file whatever.

I think the `_get_candidate_names()` only generates a random string.. so there might not be any protection against filename clashes.

Reply:
Done!